### PR TITLE
Update message when users try to update from 1st to 2nd Gen function

### PR DESCRIPTION
**Pull request by taeold**
CLI has always block the transition 1st Gen to 2nd Gen in the CLI - change here simply update the product name and point users to our public doc that documents changes necessary to transition to 2nd gen function.

Aside: Cloud Run functions doesn't allow for 1st and 2nd gen functions to share the same name.

### CODE REVIEW

File: src/deploy/functions/release/planner.ts [hunk 1]
```diff
@@ -316,9 +316,8 @@ export function checkForIllegalUpdate(want: backend.Endpoint, have: backend.Endp
 export function checkForV2Upgrade(want: backend.Endpoint, have: backend.Endpoint): void {
   if (want.platform === "gcfv2" && have.platform === "gcfv1") {
     throw new FirebaseError(
-      `[${getFunctionLabel(
-        have,
-      )}] Upgrading from GCFv1 to GCFv2 is not yet supported. Please delete your old function or wait for this feature to be ready.`,
+      `[${getFunctionLabel(have)}] Upgrading from 1st Gen to 2nd Gen is not yet supported. ` +
+        "Please review https://firebase.google.com/docs/functions/2nd-gen-upgrade before migrating to 2nd Gen.",
```
Comment (egilmorez) [inline src/deploy/functions/release/planner.ts:320]: I always feel slightly rude making this comment, but our developer style guide advises against "please."  I'd suggest the simple and direct "See <link>."

File: src/deploy/functions/release/planner.ts [hunk 2]
```diff
@@ -316,9 +316,8 @@ export function checkForIllegalUpdate(want: backend.Endpoint, have: backend.Endp
 export function checkForV2Upgrade(want: backend.Endpoint, have: backend.Endpoint): void {
   if (want.platform === "gcfv2" && have.platform === "gcfv1") {
     throw new FirebaseError(
-      `[${getFunctionLabel(
-        have,
-      )}] Upgrading from GCFv1 to GCFv2 is not yet supported. Please delete your old function or wait for this feature to be ready.`,
+      `[${getFunctionLabel(have)}] Upgrading from 1st Gen to 2nd Gen is not yet supported. ` +
```
Comment (egilmorez) [inline src/deploy/functions/release/planner.ts:319]: When do we actually display this?  On deployment of a scheduled function I guess?

**Review by milyes — APPROVED** [review]
https://github.com/firebase/firebase-tools/blob/0d5f7492511a141eb235e27fe5d0a160a61af4f5/schema/firebase-config.json

File: src/deploy/functions/release/planner.ts (hunk 2 above)
Comment (taeold) [reply src/deploy/functions/release/planner.ts:319]: On `firebase deploy` when we detect that there exists 1st Gen function whose name is same as a 2nd Gen function you have in your source.

**Review by joehan — APPROVED** [review]

**Pull request merged**